
---


# EMI Calculator


---


**About EMI**

We all have come across the term EMI in everyday life. While buying any product, we get an option of the EMI Payment method. The full form of EMI is **Equated Monthly Installment**. Through this, one must pay a specific sum to the seller on a fixed date each month with a fixed interest rate. EMI calculator helps calculate the amount with its interest rate that a buyer needs to pay to the seller.

 EMI Calculator is used to calculate home loans, car loans, bike loans, education loans EMI. In this assignment, you make you own EMI Calculator.

**Factors affecting EMI**

EMI of a loan have three major factors on which it depends:
 
* **Loan Amount:** This stands for the total amount or the principal amount that the individual has borrowed.
* **Interest Rate:** It is a rate at which the interest is charged on the amount borrowed.
* **Tenure of the Loan:** It is an agreed loan repayment time frame between the borrower and the lender.

**Formula**

E = P * r  * ( (1+r)^n) / ( (( 1 + r )^n) - 1)

 Where, E = EMI, P = Principal amount, r= rate of interest and n = Tenure in months.

---

## Problem Statement

---

1.    Let’s calculate the Home Loan EMI now using Python code. Principal Amount 50Lakhs, Rate of interest 8%, and Loan repayment tenure is 15 years. Find the EMI to be paid?
 
2.    If the interest rate increases to 10%, and you cannot pay more than 30,000 as EMI, find the maximum amount of Loan you can take/afford for the same period?

3.    For the Que 1. what is the total interest paid by you year-wise (from the 1st year to the 15th year). Also, what is the total cumulative interest paid by you in 15 years?
 
4.    For the Que 1., in how many months would you have paid atleast 50% of the Principal amount.

---

## Solutions

---

In [1]:
import math

**1.    Let’s calculate the Home Loan EMI now using Python code. Principal Amount 50Lakhs, Rate of interest 8%, and Loan repayment tenure is 15 years. Find the EMI to be paid?**

In [2]:
# Function to calculate EMI

def calc_emi(p, r, n):
  r = r / (12 * 100) # monthly interest rate
  n = n * 12 # number of months
  emi = p * r * (1 + r)**n / (((1 + r)**n) - 1)
  return emi

In [3]:
# Calculate EMI for principal amount 50 lakhs, interest rate 8%, and loan repayment tenure 15 years

p = 5000000
r = 8
n = 15

emi = calc_emi(p, r, n)
print("EMI for principal amount 50 lakhs, interest rate 8%, and loan repayment tenure 15 years:", "Rs." , emi)

EMI for principal amount 50 lakhs, interest rate 8%, and loan repayment tenure 15 years: Rs. 47782.60421651782


---

**Ans. EMI for principal amount 50 lakhs, interest rate 8%, and loan repayment tenure 15 years:** Rs. 47782.60


---


**2.    If the interest rate increases to 10%, and you cannot pay more than 30,000 as EMI, find the maximum amount of Loan you can take/afford for the same period?**

In [4]:
#function to calculate maximum loan amount

def principal(emi, r, n):
    r = r / (12 * 100) # monthly interest rate
    n = n * 12 # number of months
    loan_amount = (emi*(((1 + r)**n) - 1))/(r * (1 + r)**n)
    return loan_amount

In [5]:
# Calculate maximum loan amount for emi-30,000, interest rate 8%, and loan repayment tenure 15 years

emi=30000
r=10
n=15

loan_amount = principal(emi, r, n)
print("Maximum loan amount for EMI of 30,000, interest rate 10%, and loan repayment tenure 15 years:", "Rs.", loan_amount)

Maximum loan amount for EMI of 30,000, interest rate 10%, and loan repayment tenure 15 years: Rs. 2791723.164689835


---

**Ans. Maximum loan amount for EMI of 30,000, interest rate 10%, and loan repayment tenure 15 years:** Rs. 2791723.16

    

---


**3.    For the Que 1. what is the total interest paid by you year-wise (from the 1st year to the 15th year). Also, what is the total cumulative interest paid by you in 15 years?**

In [6]:
# Function to calculate year-wise interest paid

def calc_yearwise_interest(p, r, n, emi):
  yearwise_interest = []
  total_interest = 0
  for i in range(n*12):
    interest = p * r / (12 * 100)
    total_interest += interest
    yearwise_interest.append(interest)
    p -= emi - interest
  return yearwise_interest

In [7]:
# Calculate year-wise interest paid for principal amount 50 lakhs, interest rate 8%, and loan repayment tenure 15 years

p=5000000
r=8
n=15

yearwise_interest = calc_yearwise_interest(p, r, n, calc_emi(p, r, n))
print("Year-wise interest paid for principal amount 50 lakhs, interest rate 8%, and loan repayment tenure 15 years:", yearwise_interest)

Year-wise interest paid for principal amount 50 lakhs, interest rate 8%, and loan repayment tenure 15 years: [33333.333333333336, 33237.00486077877, 33140.03419840718, 33042.41706495311, 32944.149150609344, 32845.22611683662, 32745.643596172078, 32645.397192036442, 32544.4824785399, 32442.895000286717, 32340.630272178507, 32237.683779216248, 32134.050976300903, 32029.727288032787, 31924.708108509556, 31818.988801122836, 31712.564698353533, 31605.43110156577, 31497.583280799427, 31389.016474561307, 31279.725889614932, 31169.70670076891, 31058.95405066392, 30947.463049558228, 30835.228775111827, 30722.246272169123, 30608.51055254013, 30494.01659478028, 30378.759343968697, 30262.733711485038, 30145.934574784817, 30028.356777173263, 29909.995127577633, 29790.84440031803, 29670.8993348767, 29550.15463566576, 29428.60497179341, 29306.244976828584, 29183.06924856399, 29059.072348777627, 28934.248802992694, 28808.59310023586, 28682.099692793978, 28554.762995969155, 28426.577387832167, 28297.53

In [9]:
# Function to calculate total interest paid

def calc_interest(p, r, n, emi):
  total_interest = 0
  for i in range(n*12):
    interest = p * r / (12 * 100)
    total_interest += interest
    p -= emi - interest
  return total_interest

In [10]:
# Calculate total cumulative interest paid for principal amount 50 lakhs, interest rate 8%, and loan repayment tenure 15 years

p=5000000
r=8
n=15

total_interest = calc_interest(p, r, n, calc_emi(p, r, n))
print(f"Total cumulative interest paid for principal amount 50 lakhs, interest rate 8%, and loan repayment tenure 15 years: {total_interest:.2f}")

Total cumulative interest paid for principal amount 50 lakhs, interest rate 8%, and loan repayment tenure 15 years: 3600868.76


---

**Ans. Total cumulative interest paid for principal amount 50 lakhs, interest rate 8%, and loan repayment tenure 15 years:** 3600868.76

---

**4.    For the Que 1., in how many months would you have paid atleast 50% of the Principal amount.**

In [11]:
# Function to calculate months required to pay at least 50% of the principal

def calc_months_50_principal(p, r, n, emi):
  paid = 0
  for m in range(n*12):
    interest = p * r / (12 * 100)
    paid += emi - interest
    p -= emi - interest
    if paid >= p/2:
      return m + 1

In [12]:
# Calculate months required to pay at least 50% of the principal for principal amount 50 lakhs, interest rate 8%, and loan repayment tenure 15 years

p=5000000
r=8
n=15

months = calc_months_50_principal(p, r, n, calc_emi(p, r, n))
print("Months required to pay at least 50% of the principal for principal amount 50 lakhs, interest rate 8%, and loan repayment tenure 15 years:", months)

Months required to pay at least 50% of the principal for principal amount 50 lakhs, interest rate 8%, and loan repayment tenure 15 years: 86


---

**Ans. Months required to pay at least 50% of the principal for principal amount 50 lakhs, interest rate 8%, and loan repayment tenure 15 years:** 86

---